### ISSUE TITLE
compiling on macos 10.13.x

### ISSUE THREAD

**Issue by rev3rend** [RTcmix/RTcmix#16]
i get an error compiling from source after running 'configure'.

echo "making all in src..."
making all in src...
make -C conf
flex -l -Pconf  conf.l
make[2]: *** [lex.conf.c] Error 1
make[1]: *** [conf] Error 2
make: *** [src] Error 2

run a second time, i get this:

making all in src...
make -C conf
bison -t -p conf -d  conf.ypp
make[2]: *** [conf.tab.cpp] Abort trap: 6
make[1]: *** [conf] Error 2
make: *** [src] Error 2

**Comment by dougscott** [comment]
Does not look like an RTcmix issue, actually -- none of that code has
changed recently.  Can you try starting your compile down in src/conf/?
And if a command triggered by make fails there, try executing the command
yourself.  We need to see why for example the flex command is failing.

-Doug

**Comment by rev3rend** [comment]
okay.

the first time you 'make' in src/conf after a 'make clean' it just dies:

===
```
flex -l -Pconf  conf.l
make: *** [lex.conf.c] Error 1

```
===

if you run 'make' immediately after (still in src/conf) you get a slightly more helpful (?) error:

===
```
flex -l -Pconf  conf.l
make: *** [lex.conf.c] Error 1
10-18-61-248:conf lukedubois$ make
bison -t -p conf -d  conf.ypp
conf.ypp:25.1-12: warning: deprecated directive, use ‘%pure-parser’ [-Wdeprecated]
 %pure_parser
 ^^^^^^^^^^^^
c++ -Wno-deprecated  -I. -I/Users/lukedubois/RTcmix/include  -g3 -O3 -Wall -Wno-unused  -fno-common -DMACOSX    -m64  -DSTANDALONE -DUSE_OSX_DISPATCH -DAPPLEAUDIO -I/usr/local/opt/flex/include  -c -o conf.tab.o conf.tab.cpp
conf.tab.cpp:1148:16: error: use of undeclared identifier 'conflex'
      yychar = yylex (&yylval);
               ^
conf.tab.cpp:67:25: note: expanded from macro 'yylex'
#define yylex           conflex
                        ^
conf.ypp:59:31: error: use of undeclared identifier 'conftext'
    { (yyval.str) = my_strdup(conftext); }
                              ^
conf.ypp:63:26: error: use of undeclared identifier 'conftext'
    { (yyval.num) = atof(conftext); }
                         ^
conf.ypp:79:35: error: use of undeclared identifier 'conftext'
                        char *s = conftext + 1;    // strip quotes
                                  ^
conf.ypp:100:42: error: use of undeclared identifier 'conflineno'
   fprintf(stderr, "near line %d: %s\n", conflineno, msg);
                                         ^
5 errors generated.
make: *** [conf.tab.o] Error 1
```
===

i tried it with both apple's shipped flex/bison and the one from homebrew, which the google machine told me might be newer. same issue both times.

**Comment by dougscott** [comment]
Hm.  I build this with "Apple's shipped" versions of those things every
week for the past 10 years.  What OSX, specifically, are you running?  I
cannot take the time to figure this out right now (especially since the
RTcmix code has not changed at all recently) - I would suggest Googling
around for those flex/bison errors to see if there are flags you can add,
for now, to get past the "pure parser" issue.  Sorry!

Maybe someone else has seen this?

-Doug

**Comment by rev3rend** [comment]
thanks, doug! i'm running 10.13.6 (high sierra). i'll poke around.

**Comment by johgibso** [comment]
Hey Luke, do you have MacPorts, Homebrew, or similar installed, with perhaps different versions of flex & bison than come standard with macOS?

**Comment by johgibso** [comment]
Oh, sorry, I missed that. This %pure_parser thing is ringing a bell for me, but the bell is nearly silent. I think we have only seen perplexing things like this with Linux in recent years. I'm running Sierra at home right now, so I can't check what you might be seeing. However, I have compiled on HS at school without issues. So this is puzzling.

**Comment by dougscott** [comment]
On my machine running 10.13.6 as well:

[douglasscott:RTcmix/src/conf] dscott% sw_vers
ProductName: Mac OS X
ProductVersion: 10.13.6
BuildVersion: 17G52

[douglasscott:RTcmix/src/conf] dscott% make
flex -l -Pconf  conf.l
sh fixlexoutput.sh lex.conf.c
bison -t -p conf -d  conf.ypp
c++ -Wno-deprecated  -I. -I/Volumes/Data/Users/dscott/src/RTcmix/include -g
-O2 -Wall -Wno-unused -m64  -fno-common -DMACOSX -m64  -DSTANDALONE
-DAPPLEAUDIO   -c -o conf.tab.o conf.tab.cpp
c++ -Wno-deprecated  -I. -I/Volumes/Data/Users/dscott/src/RTcmix/include -g
-O2 -Wall -Wno-unused -m64  -fno-common -DMACOSX -m64  -DSTANDALONE
-DAPPLEAUDIO   -c -o Config.o Config.cpp
ld -r conf.tab.o Config.o -o libconf.a

[douglasscott:RTcmix/src/conf] dscott% bison --version
bison (GNU Bison) 2.3
Written by Robert Corbett and Richard Stallman.

Copyright (C) 2006 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[douglasscott:RTcmix/src/conf] dscott% flex --version
flex 2.5.35 Apple(flex-31)

**Comment by rev3rend** [comment]
super weird.

still get this the first time (and that longer error the second time):

```
flex -l -Pconf  conf.l
make[2]: *** [lex.conf.c] Error 1
make[1]: *** [conf] Error 2
make: *** [src] Error 2
```
i have the same stats as you, more or less:

```
[10-18-61-248:RTcmix lukedubois$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G4015
10-18-61-248:RTcmix lukedubois$ bison --version
bison (GNU Bison) 2.3
Written by Robert Corbett and Richard Stallman.

Copyright (C) 2006 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
10-18-61-248:RTcmix lukedubois$ flex --version
flex 2.5.35 Apple(flex-31)
10-18-61-248:RTcmix lukedubois$ cd src/conf/
10-18-61-248:conf lukedubois$ make
bison -t -p conf -d  conf.ypp
c++ -Wno-deprecated  -I. -I/Users/lukedubois/Documents/RTcmix/include  -g3 -O3 -Wall -Wno-unused  -fno-common -DMACOSX    -m64  -DSTANDALONE -DUSE_OSX_DISPATCH -DAPPLEAUDIO   -c -o conf.tab.o conf.tab.cpp
conf.tab.cpp:1279:16: error: use of undeclared identifier 'conflex'
      yychar = YYLEX;
               ^
conf.tab.cpp:629:16: note: expanded from macro 'YYLEX'
# define YYLEX yylex (&yylval)
               ^
conf.tab.cpp:63:17: note: expanded from macro 'yylex'
#define yylex   conflex
                ^
conf.ypp:59:31: error: use of undeclared identifier 'conftext'
    { (yyval.str) = my_strdup(conftext); ;}
                              ^
conf.ypp:63:26: error: use of undeclared identifier 'conftext'
    { (yyval.num) = atof(conftext); ;}
                         ^
conf.ypp:79:35: error: use of undeclared identifier 'conftext'
                        char *s = conftext + 1;    // strip quotes
                                  ^
conf.ypp:100:42: error: use of undeclared identifier 'conflineno'
   fprintf(stderr, "near line %d: %s\n", conflineno, msg);
                                         ^
5 errors generated.
make: *** [conf.tab.o] Error 1](url)
```

i'm pulling straight off the git repository. i run 'configure' at the top. anything else i might be missing?

**Comment by rev3rend** [comment]
so something's borked with flex/bison on my machine. 'lex.conf.c' (in src/conf) and 'lex.yy.c' (in src/parser/minc) are being generated as empty (0 bytes) when i run make. if i copy the files over from another computer that builds rtcmix correctly, everything builds fine. 

wonder wtf that's about.

/luke

**Comment by johgibso** [comment]
That is really weird. There's not a chance that homebrew or something would've fouled up which flex/bison was used, is there?

Anyway, glad you have a way around if for now, even though it's awkward.

J

**Comment by rev3rend** [comment]
yeah. flex uses like fifteen different other things behind your back (like the macro processor) so it's totally possible that some version of homebrew killed something.

do those parser-generated c files ever change, from computer-to-computer?

/luke

**Comment by johgibso** [comment]
I'm not sure. I know Brad used to generate those parser .c’s in RTcmix, copy them into his rtcmix~ tree, compile, and then deploy on many different Macs. But Doug would know better than me.

J

**Comment by dougscott** [comment]
The ones in src/conf change rarely. The ones the the MinC parser change pretty often. Do you have any problem compiling those?

Doug

**Comment by rev3rend** [comment]
neither will compile on my machine. as i said above, 'lex.conf.c' (in src/conf) and 'lex.yy.c' (in src/parser/minc) are being generated as empty (0 bytes) when i run make and there are flex errors as a result. if i copy over the files that are generated in a make on another machine, everything is fine.

**Comment by dougscott** [comment]
Luke, I'm going back through all the old Git issues.  Can I assume this issue was eventually solved?  Care to add a note for future reference?

**Comment by dougscott** [comment]
Luke, I'm going back through all the old Git issues.  Can I assume this issue was eventually solved?  Care to add a note for future reference?
